website/src/react-luma/components/luma-example.tsx

```diff
@@ -63,52 +68,78 @@ export const LumaExample: FC<LumaExampleProps> = (props: LumaExampleProps) => {
   let containerName = 'ssr';
 
   /** Each example maintains an animation loop */
-  const [animationLoop, setAnimationLoop] = useState<AnimationLoop | null>(null);
+  const animationLoopRef = useRef<AnimationLoop | null>(null);
+  const [canvas, setCanvas] = useState<HTMLCanvasElement | null>(null);
+  const usedCanvases = useRef(new WeakMap())
+  const currentTask = useRef<Promise<any> | null>(null);
```

Comment:
Is it possible to type this Promise stronger, instead of `any`? This part is probably the most confusing in the new code, so comments and types will help.